### PR TITLE
Add group agreements

### PR DESCRIPTION
**Pull request by nicksellen**
Implementation of https://github.com/yunity/karrot-backend/issues/385

A few things missing so far:
- try and reduce the duplication in my serializers
- ensure the membership manager can assign agreement managers
- probably initialize an agreement manager so there is an initial one
- limiting certain actions depending on agreement status (e.g. users should not be able to do pickups unless they have agreed)

Two things I think I don't need to implement (at least right now) are:
- deleting agreements
- unagreeing (gets more complicated as it might depend on what is inside the agreement content, so might need to be set whether it can be unagreed or not, per agreement, probably does not matter for now)

### CODE REVIEW

File: foodsaving/groups/api.py [hunk 1]
```diff
@@ -144,3 +145,37 @@ def modify_user_roles(self, request, pk, user_id, role_name):
...
 
         return Response(GroupMembershipInfoSerializer(instance).data)
+
+
+class IsGroupAgreementManager(BasePermission):
+    message = _('You cannot manage agreements.')
+
+    def has_permission(self, request, view):
+        group_id = request.GET.get('group')
+        print('checking agreement manager permissions', request.user, group_id)
```
Comment (tiltec) [inline foodsaving/groups/api.py]: remove print before merging

File: foodsaving/groups/serializers.py [hunk 2]
```diff
@@ -108,6 +119,64 @@ def create(self, validated_data):
         return group
 
 
+class AgreementSerializer(serializers.ModelSerializer):
```
Comment (tiltec) [inline foodsaving/groups/serializers.py]: Shouldn't this also return the users that have agreed? Can be shown in frontend to the agreement manager.

Comment (nicksellen) [reply foodsaving/groups/serializers.py]: Could do, and might be useful at some point, I just implemented the "minimum information" principle for now...

Another approach would be to include it in the membership objects in the group serializer (for membership managers), as that is where other membership management information is (e.g. roles).

Comment (tiltec) [reply foodsaving/groups/serializers.py]: Yes, that seems to be easy enough to get added later when the frontend is in development or when a user requests it. It's fine to progress without this information for now.

**Pull request merged**